qhub/template/stages/06-kubernetes-keycloak-configuration/main.tf

```diff
@@ -1,14 +1,36 @@
 resource "keycloak_realm" "main" {
-  provider = keycloak
 
   realm        = var.realm
   display_name = var.realm_display_name
+  
+  direct_grant_flow    = "direct grant"
+  enabled              = true
+  browser_flow         = "browser"
+  revoke_refresh_token = false
+  user_managed_access  = false
+  ssl_required         = "external"
+  registration_flow    = "registration"
+    
+  refresh_token_max_reuse    = 0
+  reset_credentials_flow     = "reset credentials"
+  client_authentication_flow = "clients"
+  docker_authentication_flow = "docker auth"
+
+  offline_session_max_lifespan_enabled = false
+  
+  web_authn_policy { 
+  }
+  
+  web_authn_passwordless_policy {
+  }
+
 }
 
 resource "keycloak_group" "groups" {
   for_each = var.keycloak_groups
   realm_id = keycloak_realm.main.id
-  name     = each.value
+  name     = each.key
```

Comment:
FYI `each.value` and `each.key` are equivalent when working with sets.

This will overwrite the attributes for the group and set it to empty. Is this intentional? I know you said you were using attributes for some of your components.

Reply:
To be honest, most of these changes were inspired by TF_LOG=DEBUG output from Terraform. It seemed to be getting confused about the values it was setting on each run - e.g. the difference between missing values and null values made it want to make changes that weren't really going to have any effect, so I'm just trying to set the defaults in a way that causes both terraform and mrparkers to consider them the same. So, yes, it's all intentional but basically guesswork that I believe helped in some way... I'm not too sure whether future deployments will remove all group attributes actually. We can change this in the future if it causes problems.